src/AppInstallerCLICore/Workflows/GetRequestedUrlFromManifest.cpp

```diff
@@ -0,0 +1,71 @@
+#include "pch.h"
+#include "GetRequestedUrlFromManifest.h"
+
+#include "ManifestComparator.h"
+#include "ShellExecuteInstallerHandler.h"
+
+using namespace AppInstaller::CLI;
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Manifest;
+
+namespace AppInstaller::CLI::Workflow
+{
+	void GetRequestedUrlFromManifest(Execution::Context& context)
+	{
+		if (context.Contains(Execution::Data::SearchResult)) {
+			const auto& searchResult = context.Get<Execution::Data::SearchResult>();
+			if (searchResult.Matches.size() > 1)
+			{
+				context.Reporter.Warn() << "More than one result was found please refine your query. Use --Id, --Name --Tag" << std::endl;
+			}
+			else
+			{
+				if (context.Contains(Execution::Data::Manifest)) {
+					const auto& manifest = context.Get<Execution::Data::Manifest>();
+					Workflow::ManifestComparator manifestComparator(context.Args);
+					const auto selectedLocalization = manifestComparator.GetPreferredLocalization(manifest);
+					if (context.Args.Contains(Execution::Args::Type::Homepage)) {
+						if (selectedLocalization.Homepage.empty())
+						{
+							context.Reporter.Info() << "No Homepage url found within manifest" << std::endl;
+						}
+						else {
+							if (!selectedLocalization.Homepage.empty()) {
+								if (selectedLocalization.Homepage._Starts_with("http")) {
```

Comment:
>selectedLocalization.Homepage._Starts_with("http") [](start = 12, length = 50)

See Utility::IsUrlRemote, although this brings up a great point about security around this feature.  I do think we need to do a review to ensure that we have the appropriate checks to ensure that this is in fact a valid URL.  We do currently ensure that SmartScreen is OK with it before it can get into winget-pkgs, but the client should at least do as much as we can to ensure that it isn't some form of injection attack.

Reply:
i agree, having the security check here would add additional safeguard when private/local/other Repositories. 
one question though, does [Utility::IsUrlRemote] provide security or just do this check? 
do i need to provide this check? (keep in mind i haven't looked into this as of yet. thought it would be easier to ask)

Reply:
Utility::IsUrlRemote just does the 'starts with' but for "http://" or "https://" (and not using a non-std method).

I've started a discussion internally on the security implications here.  I think the most likely result will be in hardening the parsing logic to reject the strings before they reach this part of the code.  But when we have such a check we might put it here as well just to be sure.

In the meantime, I believe that there is a good chance that the feature will need to go behind a toggle to ensure that we don't create a security issue.  If you follow the example of ```EnsureFeatureEnabled(Settings::ExperimentalFeature::Feature::ExperimentalMSStore)``` you can add one fairly easily.

Reply:
Breaking with my no comment from above :) I did add the Experimental flag to this feature as you requested. it's working on and off but my concern is that i may have missed something because the Feature and settings names were/are different. I attempted to use the same var names and the like but it appears one differs from the other. I've run a search and attempted to resolve this. it works but I would like a second pair of eyes on this new edit. Thanks